rendering/modelblock.md

```diff
@@ -0,0 +1,157 @@
+
+```java
+// snip imports and package declaration
+class BlockCustomGrass extends Block {
+    // snip a whole bunch of important stuff for brevity's sake, I'll assume you already have a basic block working.
+    public static final PropertyBool SNOWY = PropertyBool.create("snowy");
+
+    public BlockCustomGrass() {
+        super(Material.grass);
+        this.setDefaultState(this.blockState.getBaseState().withProperty(SNOWY, Boolean.valueOf(false)));
+    }
+}
+```
+
+This isn't sufficient to get the nice `snowy=true` and `snowy=false` to work in the JSON however, we're going to have to write a bit of code to tell the game what properties our block has.
+Additionally, we need to tell the game how to convert our properties in to the 4-bit metadata value.
+Since we can compute whether or not the block is snowy at runtime, we're just going to tell the game that the metadata is always 0.
+
+```java
+    @Override
+    protected BlockState createBlockState() {
+        return new BlockState(this, new IProperty[] { SNOWY });
+    }
+
+    @Override
+    public int getMetaFromState(IBlockState state) {
+        return 0;
+    }
+```
+
+We also need to be able to return the right state depending on the state of the world when rendering takes place.
+In this case, we want to know when the block above us is snow.
+
+```java
+    @Override
+    public IBlockState getActualState(IBlockState state, IBlockAccess world, BlockPos pos) {
+        Block b = world.getBlockState(pos.up()).getBlock();
+        return state.withProperty(SNOWY, Boolean.valueOf(b == Blocks.snow || b == Blocks.snow_layer));
+    }
+```
+
+And we're done!
+Everything else is handled automatically, and as long as your JSON is set up right, everything will Just Work(TM).
+Of course, sometimes we need a bit more control over what properties change the visuals of our block, and for that we turn to the default `StateMapper`'s more powerful cousin, housed at `net.minecraft.client.renderer.block.statemap.StateMap`.
+
+General StateMaps
+-----------------
+
+The most common use for the `StateMap` class is when you have a block with some `Property` that you need for other reasons but want to ignore when rendering, as it doesn't affect the visuals of the block.
+Redstone powered state is a common candidate for this sort of thing.
+For example, lets say our `BlockCustomGrass` needed to know if it was powered.
+
+```java
+class BlockCustomGrass extends Block {
+    // snip a whole bunch of important stuff for brevity's sake, I'll assume you already have a basic block working.
+    public static final PropertyBool SNOWY = PropertyBool.create("snowy");
+    public static final PropertyBool POWERED = PropertyBool.create("powered");
+
+    public BlockCustomGrass() {
+        super(Material.grass);
+        this.setDefaultState(
+            this.blockState.getBaseState().withProperty(SNOWY, boolean.valueOf(false))
```

Comment:
These should be caps Boolean, but is that call really needed? Are the values not autoboxed?